packages/plugin-ext/src/main/browser/menus/menus-contribution-handler.ts

```diff
@@ -82,6 +107,81 @@ export class MenusContributionPointHandler {
         }
     }
 
+    protected registerScmTitleAction(menuAction: Menu): void {
+        const id = this.createSyntheticCommandId(menuAction, { prefix: '__plugin.scm.title.action.' });
+        const command: Command = { id };
+        this.commands.registerCommand(command, {
+            execute: () => this.commands.executeCommand(menuAction.command)
+        });
+
+        this.scmTitleCommandRegistry.registerCommand({ command: id, when: menuAction.when });
+
+        if (menuAction.group && menuAction.group !== 'navigation') {
+            const action: MenuAction = { commandId: id };
+            this.menuRegistry.registerMenuAction([...PATH, menuAction.group], action);
+        }
+
+        this.onDidRegisterCommand(menuAction.command, pluginCommand => {
+            command.iconClass = pluginCommand.iconClass;
+            command.category = menuAction.group;
+            command.label = pluginCommand.label;
+        });
+    }
+
+    protected registerScmResourceGroupAction(action: Menu): void {
+        const id = this.createSyntheticCommandId(action, { prefix: '__plugin.scm.resource.group.action.' });
+        const command: Command = { id };
+        this.commands.registerCommand(command, {
+            execute: arg => this.commands.executeCommand(action.command, arg)
+        });
+
+        const condition = action.when;
```

Comment:
Why `when` is not used here  in `registerScmResourceAction`? There are more conditions to `scmResourceGroup`. It looks bogus.

Reply:
VsCode conditions (other then group and section) are not supported in the SCM. I think it could be another iteration.

Reply:
ok, we need an issue then